lib/livebook/settings.ex

```diff
@@ -242,4 +246,48 @@ defmodule Livebook.Settings do
   defp broadcast_env_vars_change(message) do
     Phoenix.PubSub.broadcast(Livebook.PubSub, "settings", message)
   end
+
+  @doc """
+  Sets default file system.
+  """
+  @spec set_default_file_system(file_system_id()) :: :ok
+  def set_default_file_system(file_system_id) do
+    storage().insert(:settings, "global", default_file_system_id: file_system_id)
+  end
+
+  @doc """
+  Settings default file system
+  """
+  def default_file_system do
+    case storage().fetch_key(:settings, "global", :default_file_system_id) do
+      :error ->
+        Livebook.Config.local_filesystem_home()
+
+      {:ok, default_file_system_id} ->
+        {_id, file} =
+          Enum.find(Livebook.Settings.file_systems(), fn {id, _file} ->
+            id == default_file_system_id
+          end)
+
+        FileSystem.File.new(file)
+    end
```

Comment:
Here we should return the actual filesystem, then let's add `default_file_system_home` that does `File.new(default_file_system())` :)

Note that we can use `default_file_system_id()` that we have below and find the filesystem based on that.